sdk/go/common/resource/plugin/plugin_test.go

```diff
@@ -89,3 +97,92 @@ func TestParsePort(t *testing.T) {
 		require.Error(t, err)
 	}
 }
+
+func TestHealthCheck(t *testing.T) {
```

Comment:
Does this check reproduce https://github.com/pulumi/pulumi/issues/18238?

Reply:
Last time I looked we didn't have a repro for 18238, it was based on log analysis. Unfortunately. I think absent a real repro this PR is a solution that's the best we can do.